### PR TITLE
fix(linter): convert tslint:disable comments as part of project conversion

### PR DESCRIPTION
**Pull request by JamesHenry**
## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/linter/src/utils/convert-tslint-to-eslint/convert-to-eslint-config.ts [hunk 1]
```diff
@@ -188,3 +189,49 @@ export async function convertToESLintConfig(
...
+  /**
+   * We need to avoid a direct dependency on tslint-to-eslint-config
+   * and ensure we are only resolving the dependency from the user's
+   * node_modules on demand (it will be installed as part of the
+   * conversion generator).
+   */
+  const { convertFileComments } = getConvertToEslintConfig();
+
+  const { root } = readProjectConfiguration(tree, projectName);
+  const allSourceFiles = allFilesInDirInTree(tree, root);
```
Comment (FrozenPandaz) [inline packages/linter/src/utils/convert-tslint-to-eslint/convert-to-eslint-config.ts:223]: We can use [`visitNotIgnoredFiles`](https://github.com/nrwl/nx/blob/01967f89b1b281750ae74c742ac2be12aa881c5e/packages/devkit/src/generators/visit-not-ignored-files.ts#L10)

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: packages/linter/src/utils/convert-tslint-to-eslint/convert-to-eslint-config.ts [hunk 2]
```diff
@@ -188,3 +194,47 @@ export async function convertToESLintConfig(
     ),
   };
 }
+
+function likelyContainsTSLintComment(fileContent: string): boolean {
+  return fileContent.includes('tslint:');
+}
+
+function allFilesInDirInTree(tree: Tree, dir: string): string[] {
```
Comment (FrozenPandaz) [inline packages/linter/src/utils/convert-tslint-to-eslint/convert-to-eslint-config.ts:202]: This is unused now and also `join` and `normalizePath`.

**Review by FrozenPandaz — APPROVED** [review]

**Comment by pburkindine** [comment]
@FrozenPandaz @JamesHenry are there any special instructions so that this migration will be performed?  I ran `nx migrate latest` this morning and it didn't change anything tslint-wise.

**Comment by JamesHenry** [comment]
@pburkindine It's not a migration in the sense of `nx migrate`. It is not up to Nx to decide when/if users should move from TSLint to ESLint, it needs to be their choice. Therefore, it is a regular Nx generator which is invoked on demand by users, just like when generating apps, libs etc.

You can either trigger it by searching for `convert-tslint-to-eslint` in Nx Console in VSCode, or by accessing it from the specific package collection. E.g. for Angular projects:

```sh
npx nx g @nrwl/angular:convert-tslint-to-eslint
```

And then follow the prompts from there

**Pull request merged**